### ISSUE TITLE
Compiler FE: Introduce target configuration

### ISSUE THREAD

**Issue by mhs4670go** [Samsung/ONE#12973]
## What

Let's introduce target configuration feature.

Target configuration applies to specific core in the system. For instance, if NPU has two cores: npu0, npu1, the target option will be used like `--target npu0` or `--target npu1`.

When users try to support multiple backends, they should write dedicated configuration files per the backends. But, the contents of the file from importing the model to optimizing it could be almost same since the only difference would be from the codegen things. Also, it's kinda tricky to maintain so many backend properties for the backend tools: `one-codegen`, `one-infer` and `one-profile`. And it's also hard to align such parameters among the tools, which can be resolved if they see the same properties or a file that contains them.

So, I'd like to introduce target configuration feature for resolving these discomfort.

`target` key will be added to the configuration file. Then, `onecc` automatically will add keys to the keys of `one-cmds` tools and add `--target ${TARGET}` to the backend commands. 

```ini
[onecc]
one-import-tf=True
one-optimize=True
one-codegen=True
target=npu0
```

Adding `target` key to the _onecc_ section is same with belows.

```ini
[one-import-tf]
# ..
target=npu0
[one-optimize]
# ..
target=npu0
[one-codegen]
# ..
command=${COMMAND} --target command
```

Simply put, users just need to do two things.
- Write target configuration file to the `/usr/share/one/target/${TARGET_NAME}.${EXT}`
- Add `target` key to the _onecc_ section

### TODO

- [ ] Decide file format of the target configuration file. (e.g. ini, json, etc)
- [ ] Decide file content template or hierarchy.

**Comment by lemmaa** [comment]
@mhs4670go , thank you for your suggestion.

I have two questions.
1. What happens if there is a component that does not support the `--target ${TARGET}` option?
2. In the case of 'target=xxx', the characteristics seem to be different from other keys in [onecc] section. How about introducing a separate section like [common-options]?

**Comment by mhs4670go** [comment]
@lemmaa 

> What happens if there is a component that does not support the `--target ${TARGET}` option?

I think that case users'd better just not use `target` option. 

> In the case of 'target=xxx', the characteristics seem to be different from other keys in [onecc] section. How about introducing a separate section like [common-options]?

Sounds a good idea:)
